driver-core/src/main/java/com/datastax/driver/core/KeyspaceMetadata.java

```diff
@@ -262,21 +286,61 @@ public String exportAsString() {
 
         sb.append(asCQLQuery()).append('\n');
 
-        for (UserType udt : userTypes.values())
+        // include types, tables, views, functions and aggregates, each ordered by name, with one small exception
+        // being that user types are ordered topologically and then by name within same level.
+        for (UserType udt : getSortedUserTypes())
```

Comment:
Actually it does appear to be relevant.   When connecting and building metadata, if a UDT depends on another UDT, it needs to be built in order, otherwise:

```
com.datastax.driver.core.exceptions.UnresolvedUserTypeException: Cannot resolve user type unresolved_user_type_test."C"
	at com.datastax.driver.core.DataTypeCqlNameParser.parse(DataTypeCqlNameParser.java:149)
	at com.datastax.driver.core.DataTypeCqlNameParser.parse(DataTypeCqlNameParser.java:120)
	at com.datastax.driver.core.UserType.build(UserType.java:93)
	at com.datastax.driver.core.SchemaParser.buildUserTypes(SchemaParser.java:196)
	at com.datastax.driver.core.SchemaParser.buildKeyspaces(SchemaParser.java:127)
	at com.datastax.driver.core.SchemaParser.refresh(SchemaParser.java:64)
	at com.datastax.driver.core.ControlConnection.refreshSchema(ControlConnection.java:336)
	at com.datastax.driver.core.ControlConnection.tryConnect(ControlConnection.java:276)
	at com.datastax.driver.core.ControlConnection.reconnectInternal(ControlConnection.java:201)
	at com.datastax.driver.core.ControlConnection.connect(ControlConnection.java:79)
	at com.datastax.driver.core.Cluster$Manager.negotiateProtocolVersionAndConnect(Cluster.java:1600)
	at com.datastax.driver.core.Cluster$Manager.init(Cluster.java:1518)
	at com.datastax.driver.core.Cluster.getMetadata(Cluster.java:399)
	at com.datastax.driver.core.UnresolvedUserTypeTest.should_resolve_nested_user_types(UnresolvedUserTypeTest.java:114)
```